### PR TITLE
ENG-9919, use uniqueId to match sentinel and first MP fragment on replay.

### PR DESCRIPTION
**Pull request by luyangco**
In ENG-9779, we don't reuse txnId of MP fragments from CL on replay to avoid
interleaving txnId for MP fragments and SP txn. But ReplaySequencer checks to
see if txnId matches between sentinel and MP fragments before it releases MP
fragments, so here we use uniqueId instead to match the sentinel and fragments.

Change-Id: I70fe249471a55413e7ebd6be827460d5788a3454

### CODE REVIEW

File: src/frontend/org/voltdb/iv2/MpInitiatorMailbox.java [hunk 1]
```diff
@@ -282,7 +282,7 @@ private void repairReplicasWithInternal(List<Long> needsRepair, VoltMessage repa
         if (repairWork instanceof Iv2InitiateTaskMessage) {
             Iv2InitiateTaskMessage m = (Iv2InitiateTaskMessage)repairWork;
             Iv2InitiateTaskMessage work = new Iv2InitiateTaskMessage(m.getInitiatorHSId(), getHSId(), m);
-            m_scheduler.updateLastSeenTxnIds(work);
+            m_scheduler.updateLastSeenUniqueIds(work);
```
Comment (nshi) [inline src/frontend/org/voltdb/iv2/MpInitiatorMailbox.java]: This one is for MPI failure resolution, not for replay. Don't change this.

Comment (luyangco) [inline src/frontend/org/voltdb/iv2/MpInitiatorMailbox.java]: I think it's harmless to leave it there, actually `updateLastSeenTxnIds()` does nothing in repair path, only replay and drv1 path cares `updateLastSeenTxnIds()`

File: src/frontend/org/voltdb/iv2/SpScheduler.java [hunk 2]
```diff
@@ -305,20 +305,27 @@ public boolean sequenceForReplay(VoltMessage message)
...
-            sequenceWithTxnId = ((TransactionInfoBaseMessage)message).getOriginalTxnId();
+            VoltDB.crashLocalVoltDB("DRv1 path should never be called", true, null);
+            sequenceWithUniqueId = ((TransactionInfoBaseMessage)message).getOriginalTxnId();
         }
+        //--------------------------------------------
 
         if (sequenceForReplay) {
-            InitiateResponseMessage dupe = m_replaySequencer.dedupe(sequenceWithTxnId,
+            //--------------------------------------------
+            // DRv1 path, mark for future removal
```
Comment (nshi) [inline src/frontend/org/voltdb/iv2/SpScheduler.java]: Command log replay also uses this path.

**Comment by nshi** [comment]
Replay sequencer tests are broken. http://ci.voltdb.lan/view/Branch-jobs/view/branch-ENG-9919/job/branch-2-pro-junit-ENG-9919/lastCompletedBuild/testReport/

File: src/frontend/org/voltdb/iv2/SpScheduler.java (hunk 2 above)
Comment (luyangco) [inline src/frontend/org/voltdb/iv2/SpScheduler.java]: I'll remove the comment here.

**Pull request merged**